### DISCUSSION TITLE
RP2040 multiple bulk out transfers failing

### DISCUSSION THREAD

**Discussion by ReFil** [adafruit/Adafruit_TinyUSB_Arduino#275] [Q&A]
Hi, I'm working on a project that requires sending lots of bulk transfers one after the other to the same endpoint in a vendor specific application, I've got communication with the USB downstream device, i can read the descriptor and take a single 16 byte bulk in transfer, but when i run the bulk out transfer the first 4 byte tuh_edpt_xfer returns 1, and every subsequent one returns 0. I think it might be connected to blocking as despite trying to set complete_cb to both 0 and NULL on different occasions the blocking doesn't seem to be working. 

After all the bulk out transfers i'm also trying a control transfer and that seems to cause the program to lock up until the device is disconnected and then the function tuh_control_xfer returns 1. There doesn't seem to be any mechanism to open the control endpoint, is this because it's already handled in the background?

**Comment by ReFil** [comment]
done more testing and even subsequent transfers as small as 4 bytes fail. I'm going to get my logic analyser on the lines this weekend and see if i can find a root cause

**Comment by hathach** [comment]
can you provide your setup details such board, this library version, bsp core version etc ... host vendor may have race condition, try to bump up version and re-testing it again.

**Reply by ReFil** [reply]
Genuint pi pico board was using v 2.0.2 library
